frontend/src/shared/hooks/useDownload.ts

```diff
@@ -53,13 +50,14 @@ const useDownload = (): UseDownload => {
     ): Promise<void> => {
         setIsLoading(true);
         try {
-            await download(id, type, watchType).then((response: any) => {
-                // response.data.file is an array of bytes
-                const blob = b64toBlob(response.data.file, "application/octet-stream", 512);
+            const response = await download(id, type, watchType);
+            // response.file is an array of bytes
+            if (response.file) {
+                const fileBlob = b64toBlob(response.file, "application/octet-stream", 512);
                 setIsDownloading(true);
-                saveAs(blob, `${watchType} ${id}.zip`);
-            });
-            setErrorState(null);
+                setErrorState(null);
+                saveAs(fileBlob, `${watchType} ${id}.zip`);
+            }
```

Comment:
This is a nitpick, feel free to ignore this.

I think its cleaner and helps readability if we make use of guard clauses.
```
            const response = await download(id, type, watchType);
            // response.file is an array of bytes
            if (!response.file) {
                return;
            }

            const fileBlob = b64toBlob(response.file, "application/octet-stream", 512);
            setIsDownloading(true);
            setErrorState(null);
            saveAs(fileBlob, `${watchType} ${id}.zip`);
```